### PR TITLE
EXRLoader: support HalfFloatType output

### PR DESCRIPTION
**Pull request by WestLangley**
Usage:
```js
new THREE.EXRLoader()
	.setType( THREE.FloatType ) // alt: THREE.HalfFloatType
	.load( 'file.hdr', function ( texture, textureData ) { ...
```

### CODE REVIEW

File: examples/js/loaders/EXRLoader.js [hunk 1]
```diff
@@ -76,11 +76,19 @@
 THREE.EXRLoader = function ( manager ) {
 
 	this.manager = ( manager !== undefined ) ? manager : THREE.DefaultLoadingManager;
+	this.type = THREE.FloatType;
 
 };
 
 THREE.EXRLoader.prototype = Object.create( THREE.DataTextureLoader.prototype );
 
+THREE.EXRLoader.prototype.setType = function ( value ) {
```
Comment (Mugen87) [inline examples/js/loaders/EXRLoader.js]: You might want to add the new method `setType()` and the property `type` to the TS declaration file.

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
Should we eventually rename these `FloatType`, `HalfFloatType`, etc to `FloatFormat` or `FloatFormatType`? Other suggestions? Just `Type` sounds too generic...

**Comment by WestLangley** [comment]
`FloatType` is a `TextureDataType` which is assigned to `Texture.type`.

Format is used for `RGBA`.

**Comment by WestLangley** [comment]
We could rename `.setType()` to `.setOutputType()` if you feel strongly about it.

**Comment by mrdoob** [comment]
Just mentioned it because we're already using that property for [something else](https://github.com/mrdoob/three.js/blob/dev/src/materials/LineBasicMaterial.js#L22). And I think we should remove that one too anyway...

So would `.setDataType()`, `FloatDataType`, `HalfFloatDataType`, ... work?

**Comment by WestLangley** [comment]
Hmm...
```js
this.type = 'LineBasicMaterial';
```
Yes, `.type` is poor property name. Why not use `object.constructor.name` instead?

```js
scene.traverse( function( node ) { console.log( node.type, node.constructor.name ) } );

Scene Scene
Mesh Mesh
CubeCamera CubeCamera
PerspectiveCamera PerspectiveCamera
Mesh Mesh
AmbientLight AmbientLight
PointLight PointLight
SpotLight SpotLight
LineSegments CameraHelper // oops, well than can be fixed...
Mesh Mesh
```

>So would .setDataType() ... work?

Yes. Or `.setOutputDataType()` to be more clear.

I'd avoid changing the the constant names `THREE.FloatType`, etc.,  if possible.

**Comment by mrdoob** [comment]
> Yes, `.type` is poor property name. Why not use `object.constructor.name` instead?

I think at the time some browser didn't support that... But yes, I'll look into cleaning that up.

>>So would .setDataType() ... work?
>
> Yes. Or `.setOutputDataType()` to be more clear.

That sounds much better indeed!

**Comment by WestLangley** [comment]
@mrdoob Actually, I prefer your suggestion: `setDataType()`. Is that OK with you?

**Comment by mrdoob** [comment]
Yep!

**Pull request merged**